### PR TITLE
Add compatibility matrix for Jet and IMDG

### PR DESCRIPTION
**Pull request by gurbuzali**
Fixes https://github.com/hazelcast/hazelcast-jet/issues/1468

Checklist:
- [ ] Labels and Milestone set
- [ ] Added a line in `hazelcast-jet-distribution/src/root/release_notes.txt` (for any non-trivial fix/enhancement/feature)
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] Updated `examples/README.md` (when adding a new code sample)

### CODE REVIEW

File: site/docs/api/sources-sinks.md [hunk 1]
```diff
@@ -899,6 +899,26 @@ p.readFrom(Sources.remoteMap("inputMap", cfg));
 ...
 ```
 
+#### Compatibility
+
+When reading or writing to remote sources, Jet internally creates a
+client. This client uses the embedded IMDG version to connect to the
+remote cluster. Starting with Hazelcast 3.6, Hazelcast server & client
+versions are backward and forward compatible within the same major
+version.
```
Comment (viliam-durina) [inline site/docs/api/sources-sinks.md:908]:
```suggestion
When reading or writing to remote IMDG cluster, Jet internally creates a
Hazelcast client. This client has the same version as is the IMDG
embedded with Jet. Starting with Hazelcast 3.6, Hazelcast server &
client versions are backward and forward compatible within the same
major version.

The table below summarizes the IMDG versions embedded in each Jet version:
```

File: site/docs/api/sources-sinks.md [hunk 2]
```diff
@@ -899,6 +899,26 @@ p.readFrom(Sources.remoteMap("inputMap", cfg));
...
+
+|Jet Version|Embedded IMDG Version|CompatibleVersions|
+|:-----|:------------------|:-----------|
+|Jet 3.0|Hazelcast 3.12|Hazelcast 3.x.x|
+|Jet 3.1    |Hazelcast 3.12.1   |Hazelcast 3.x.x|
+|Jet 3.2    |Hazelcast 3.12.3   |Hazelcast 3.x.x|
+|Jet 3.2.1  |Hazelcast 3.12.5   |Hazelcast 3.x.x|
+|Jet 3.2.2  |Hazelcast 3.12.6   |Hazelcast 3.x.x|
```
Comment (viliam-durina) [inline site/docs/api/sources-sinks.md:916]:
```suggestion
|Jet 3.1    |Hazelcast 3.12.1   |Hazelcast 3.6+.x|
|Jet 3.2    |Hazelcast 3.12.3   |Hazelcast 3.6+.x|
|Jet 3.2.1  |Hazelcast 3.12.5   |Hazelcast 3.6+.x|
|Jet 3.2.2  |Hazelcast 3.12.6   |Hazelcast 3.6+.x|
```

**Review by viliam-durina — APPROVED** [review]

File: site/docs/api/sources-sinks.md (hunk 2 above)
Comment (olukas) [reply site/docs/api/sources-sinks.md:916]: Isn't just `3.6+` better than `3.6+.x`?

File: site/docs/api/sources-sinks.md [hunk 3]
```diff
@@ -899,6 +899,26 @@ p.readFrom(Sources.remoteMap("inputMap", cfg));
...
+|Jet 4.0    |Hazelcast 4.0      |Hazelcast 4.x.x|
```
Comment (olukas) [inline site/docs/api/sources-sinks.md:917]: `Hazelcast 4.x.x` looks quite weird for me, should not we use just `4.x`? or at least `4.x.y` or `4.y.z`?

**Pull request merged**